node/server.ts

```diff
@@ -0,0 +1,155 @@
+import * as util from 'util';
+import * as child_process from 'child_process';
+import * as Git from 'nodegit';
+import * as express from 'express';
+import { c } from './lib/Log';
+import { hbs } from './lib/Hbs';
+import __rootDir, { __klausDir, __nodeDir } from './lib/RootDirFinder';
+import { Utils } from './lib/Utils';
+import { Repo } from './app/Repo';
+import {
+	indexTree,
+	indexBlob,
+	rawBlob,
+	blameBlob,
+	viewCommit,
+	historyCommits,
+} from './app/routes';
+const __exec = util.promisify(child_process.exec);
+
+const app = express();
+const PORT = process.env.PORT || 8888;
+
+
+// Express setup
+app.set('views', `${__nodeDir}/views`);
+app.set('view engine', 'hbs');
+app.engine('hbs', hbs.renderFile.bind(hbs));
+
+app.use(
+	'/static',
+	express.static(`${__klausDir}/static`)
+);
+app.get(
+	'/favicon.ico',
+	(req, res) => res.sendFile(`${__klausDir}/static/favicon.png`)
+);
+
+/**
+ * Note: we only support branch names and tag names
+ * not containing a `/`.
```

Comment:
I see where you're coming from but at least in the software engineering world feature branches (`feature/xyz`) are very common.

Reply:
Yes – for a v1 for the sake of simplicity I think this will work but we can always add support later